pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,201 @@
+package logger
+
+import (
+	"container/list"
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+
+var UnixErrorLogPath = "/opt/aws/aws-otel-collector/logs/aws-otel-collector-error.log"
+
+var WindowsErrorLogPath = "C:\\ProgramData\\Amazon\\AWSOTelCollector\\Logs\\aws-otel-collector-error.log"
+
+var errorLogfile = getErrorLogFilePath()
+
+var errorLogMaxAge = 10
+
+var queue = list.New()
+var errorLogChannel = make(chan zapcore.Entry, 10)
+var ticker = time.NewTicker(10 * time.Minute)
```

Comment:
Use env var STATUS_MESSAGE_TTL to set the frequency of the ticker. It should either be the same as STATUS_MESSAGE_TTL or should be less

Reply:
sure